### ISSUE TITLE
Mulitplayer is now officially broken (GG CLASSIC)

### ISSUE THREAD

**Issue by Duchenkuke** [TheGameCreators/GameGuruRepo#1682]
A while ago, it was possible to have amazing online matches with fellow gameguru members on awesome maps.

For weeks I have been trying to get back to it, always without success.
Ive had multiple members of the community testing it with me now and it never worked.

I would like to ask the TGC Staff if something changed regarding multiplayer.
It seems like a server issue of some kind.

I wrote a forum thread on this topic. Please check it out:
https://forum.game-guru.com/thread/223217

**Comment by plemsoft** [comment]
@LeeBamberTGC Was not able to find any reason for this, checked all code , there is no changes in the classic version from many year back , so nothing changed, so could be a change in steam ?

**Comment by LeeBamberTGC** [comment]
@plemsoft Yes I do not remember any multiplayer work being done in 2021.

**Comment by Duchenkuke** [comment]
its very strange. I think its something with the steam workshop of how gg handles it.

**Comment by C4ever08** [comment]
Yes multiplayer need's to be looked at. It defiantly worked at some point during last year since then when you try to get a match going it will prompt saying its downloading files but its not doing anything and you have to go out of Game Guru and load back in.
I think me And Duchenkuke had it work once but there were no assets in map that should have been there. And ever since it just wont work. this does seem to be a Steam workshop issue as you should see GG download the files needed but instead like I said it just doesn't. I hope you guys can look into this it would be really appreciated thanks.

**Comment by LeeBamberTGC** [comment]
@Duchenkuke Classic Multiplayer officially unbroken for the March Fixes. I will upload as BETA and PP soon so you can test it out. Turns out a few lines got wiped out, not sure when but probably a good while back during some VRQ work:

![image](https://user-images.githubusercontent.com/35840583/157569529-1c7b93d0-3c0b-45ba-aa0d-0043af633fb8.png)

Also found the baseList had not been updated for years. Basically, the new one ensures no stock media is included in the workshop item meaning quicker collection, uploading and downloading when custom content is found :)

**Comment by Duchenkuke** [comment]
What a message to start the day, damn Thank you so much. I have an entire discord only for GG Multiplayer related stuff, they are gonna be so happy!!

**Comment by LeeBamberTGC** [comment]
@Duchenkuke I am still seeing some strangeness with Workshop items. It works great in one direction but not the other. Currently investigating this:

https://help.steampowered.com/en/faqs/view/71D3-35C2-AD96-AA3A

Apparently, it allows Steam to block access to workshop content, and yet does not produce a specific error except a generic one which is what was happening with the instant subscribe failure.

**Comment by Duchenkuke** [comment]
Okay. Should be able to test again this weekend when I got my day off. Would be cool if we find the issue!

**Comment by LeeBamberTGC** [comment]
@Duchenkuke Ran some more tests last night and true enough custom assets encrypt, save as workshop item, subscribed and downloads fine in one direction:

![image](https://user-images.githubusercontent.com/35840583/157668414-cb964e17-fbf9-44e6-be2f-f0d43ae6eef4.png)


When try the other way and getting a new Steam account and fresh install (NUC) to subscribe, it fails right away.  It can send workship data fine, just not receive it. I suspect firewall/protection on a fresh Windows install (or similar) but right now the request to download files quite almost right away with a generic failure error code. Strange.

**Comment by plemsoft** [comment]
@LeeBamberTGC "we restrict access to certain community features until an account has spent at least $5.00 USD in Steam." :)

**Comment by LeeBamberTGC** [comment]
@plemsoft Yes I added £10 to my steam wallet this morning, no joy. Still fails to subscribe. Once March Update and EA is out the way, I want to have a go at updating the Steam API, it might simply be some new server stuff at their end and we need to update our Steam API. Another possibility is the subscribe needs to happen first and be complete before the download attempt is made. Do not worry about this now, we can come back to it once VRQ and MAXEA are behind us :)

**Comment by Duchenkuke** [comment]
May this helps ( I dont know )

https://www.youtube.com/watch?v=fQHQZhyyNFo&t=182s
this was the last match I recorded . And this was the last time we were able to play MP.
THis was in January 2021. So it was working back then. Something changed in the mean time, could it be something with gg max ?

**Comment by Duchenkuke** [comment]
okay we aborted the lobby when we hit the 30 minute mark.
Now we are trying a stock MP map (one that comes with gg) it uses STOCK ASSETS only.
Will report how it goes.

**Comment by Duchenkuke** [comment]
fun fact: there was someone else testing MP when we tested it :D
![123](https://user-images.githubusercontent.com/35990507/171036848-572cb35e-852a-486e-8c6c-08c2f56ef61d.JPG)

![121](https://user-images.githubusercontent.com/35990507/171036861-55544035-cd31-4e06-a6b0-cb0fae99e22f.JPG)

now we are 3 guys in the lobby, 1 random dude joined^^

OKAY I can confirm that MP on STOCK MAPS works !

**Comment by Duchenkuke** [comment]
summary:

MP works with stock maps (in our test it crashed after like 10 minutes)
MP doesnt work with custom maps.

If this could be adressed, it would be awesome.

**Comment by LeeBamberTGC** [comment]
ALL: As hinted at during our recent live broadcasts, the decision has been made to avoid fixing this wheel, as we will be replacing the whole vehicle soon.  The current system uses Steamworks, and this bug required a big update to get custom media working again. The new system will be Photon, which has a different way to share custom content that is working well with our sister product VR Quest (the educational non-violent version of the Classic engine).

The larger scope of this is that 'maybe' if the implementation for Classic goes well, we can use this as the basis of adding multiplayer to GameGuru MAX.  Photon has impressed us over the years, and with additional options to allow users to create their own servers and of course the ability to share your multiplayer games with friends without them needing Steam or a copy of GameGuru Classic, it feels like the right use of available energy :)

Keeping this as a hybrid bug/feature as both are technically true here.

**Comment by Teabone3** [comment]
Any progress?

**Comment by Teabone3** [comment]
No problem from me, I'm just asking on behalf of others that brought it up last week.  I'll nudge the group to see if any of them can provide further context and tests.

**Comment by Duchenkuke** [comment]
For the Protocol: Nothing has changed and it still doesnt work. For over 4 years now
